fwidgen/src/main.rs

```diff
@@ -172,13 +197,17 @@ fn main() -> Result<()> {
         }
     };
 
-    let mut digest = Sha3_256::new();
+    let mut digest = args.digest.to_dyn_digest();
     digest.update(&image);
-    digest.update(vec![0xff; pad]);
+    digest.update(vec![0xff; pad].as_ref());
 
     let digest = digest.finalize();
+    let name = args
+        .digest
+        .to_possible_value()
+        .ok_or(anyhow!("Invalid digest"))?;
 
-    println!("{}", hex::encode(digest));
+    println!("{};{}", name.get_name(), hex::encode(digest));
```

Comment:
I'm not a clap expert but this feels a little backwards given we've already done the parsing with clap and it shouldn't be possible to get an invalid digest.

Reply:
Yeah this is a bit weird. The choice I had to make was whether to provide an implementation of `Display` for `enum Digest` duplicating the work that clap / ValueEnum already does, or use the clap interface to get the string representation (`to_possible_value`). I went with the later mostly because I want the string passed via `--digest` to be the same one that ends up in the output and I didn't want to create a situation where they could get out of sync.

To your point: clap has already parsed the digest string by the time we get here so AFAIK `to_possible_value()` will never return None. I went back and forth between the `anyhow!("Invalid digest")` and `unreachable!()`. Would you prefer `unreachable()` or do you want me to stay away from the clap / `ValueEnum` interface all together?

Reply:
I think I'd like to get an opinion from one of our local clap experts if only to make sure we aren't going to hit any surprises. If nobody there has any issue or this is actually a really common idiom I'm okay with it.

Reply:
spoke with @lzrd last night & he agrees that implementing `Display` will be more clear & that once clap is done parsing args we shouldn't rely on it for other purposes. I'll flip this back over and update the PR.